src/components/Dialogs/BoardDialog.vue

```diff
@@ -101,22 +103,24 @@ const formData = ref<{ name: string; columns: string[] }>({
 })
 const formNameError = ref(false)
 
-const submit = async () => {
-  if (formData.value.name === '') return
-
-  emits('close-dialog')
-
-  if (props.action === 'add') {
-    await addNewBoard({
+const submitFunctions = {
+  add: () =>
+    addNewBoard({
+      userID: userID.value,
+      name: formData.value.name,
+      columns: formData.value.columns.map((name) => ({ name, tasks: [] }))
+    }),
+  edit: () =>
+    editBoard({
       name: formData.value.name,
       columns: formData.value.columns.map((name) => ({ name, tasks: [] }))
     })
-    return
-  }
+}
 
-  await editBoard(
-    formData.value.name as string,
-    formData.value.columns as string[]
-  )
+const submit = () => {
+  if (formData.value.name === '') return
+
+  emits('close-dialog')
+  submitFunctions[props.action as keyof typeof submitFunctions]()
```

Comment:
It should be handeled with vee-validate, but that's not in scope of this PR. Anyway it would be good to add a comment like so:
```js
// TODO: this form should be handled with vee-validate

Reply:
I would add this comment but I'm not really sure if there's a purpose of applying `vee-validate` here, since the only thing that needs to be checked here is if this input value isn't empty. As you can see, I can make it work with just one line of code without `vee-validate`.

Also, applying `vee-validate` for this would be troublesome because `vee-validate` doesn't allow me to display errors **inside** the inputs, in accord to my mock-up.